Modules/_zstd/zstddict.c

```diff
@@ -60,24 +59,28 @@ _zstd_ZstdDict_new_impl(PyTypeObject *type, PyObject *dict_content,
     }
 
     /* Check dict_content's type */
-    self->dict_content = PyBytes_FromObject(dict_content);
-    if (self->dict_content == NULL) {
+    if (dict_content == NULL) {
```

Comment:
It is never NULL.

BTW, overriding arbitrary exception with a TypeError in the current code is wrong.